kalliope/signals/order/order.py

```diff
@@ -121,9 +129,27 @@ def waiting_for_trigger_callback_thread(self):
             self.trigger_instance.pause()
         else:
             Utils.print_info("Waiting for trigger detection")
+
         # this loop is used to keep the main thread alive
-        while not self.trigger_callback_called:
-            sleep(0.1)
+        if self.settings.variables["max_time"] == 0:
```

Comment:
what is "max-time" ? Where is the doc? What if the flag is missing from the settings?

Reply:
max-time is the user defined time the trigger should remain active after use. It should be in settings.yml but I do not know enough python to implement it properly. Therefore, I implemented it as a user variable, a blatant hack.

Reply:
Need to be somewhere in the doc.
And need to be loaded like other variables in the SettingLoader.

Reply:
> what is "max-time" ? Where is the doc? _What if the flag is missing from the settings?_

  File "/usr/local/lib/python3.8/dist-packages/kalliope-0.7.0-py3.8.egg/kalliope/signals/order/order.py", line 134, in waiting_for_trigger_callback_thread
    if self.settings.variables["max_time"] == 0:
KeyError: 'max_time'

Also, IMHO it should be an setting option and not a variable.

Reply:
It should be a setting option that is then placed in a variable via the setting loader like all other settings.

Reply:
I agree, please refer to my CHANGELOG. I do not have the knowledge or skill to implement the time variables as application settings.

Reply:
Well, I can close the PR then?

Reply:
BTW like I mentioned you've deleted all the previous content of the changelog.

Reply:
And for the variable you woulkd only have to copy the behavior [of existing var here](https://github.com/kalliope-project/kalliope/blob/master/kalliope/core/ConfigurationManager/SettingLoader.py#L53)